packages/catalog-server/src/lib/graphql.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {readFile} from 'fs/promises';
+import {createRequire} from 'module';
+import {makeExecutableSchema} from '@graphql-tools/schema';
+import {isReadablePackage, isReadablePackageVersion, PackageInfo, Resolvers} from '@webcomponents/catalog-api/lib/schema.js';
+import {Catalog} from './catalog.js';
+
+const require = createRequire(import.meta.url);
+
+/**
+ * Creates a GraphQLSchema object with resolvers
+ */
+export const makeExecutableCatalogSchema = async (catalog: Catalog) => {
+  const schemaPath = require.resolve(
+    '@webcomponents/catalog-api/src/lib/schema.graphql'
+  );
+  const schemaSource = await readFile(schemaPath, 'utf8');
+
+  // Resolvers for our schema. See https://graphql.org/learn/execution/ for
+  // an explaination of the role of resolvers in performing GraphQL queries.
+  const resolvers: Resolvers = {
+    Query: {
+      async package(_parent, {packageName}: {packageName: string}): Promise<PackageInfo | null> {
+        console.log('query package', packageName);
+        const packageInfo = await catalog.getPackageInfo(packageName);
+        if (packageInfo === undefined) {
+          console.log(`package ${packageName} not found in db, importing`);
+          let result;
+          try {
+             result = await catalog.importPackage(packageName);
+          } catch (e) {
+            console.error(e);
+            throw e;
+          }
+          const {packageInfo} = result;
+          if (isReadablePackage(packageInfo)) {
+            console.log(`package ${packageName} is readable`);
+            return packageInfo;
+          } else {
+            console.log(`package ${packageName} is not readable`);
+            return null;
+          }
+        } else if (isReadablePackage(packageInfo)) {
+          console.log(`package ${packageName} found in db and readable`);
+          return packageInfo;
+        } else {
+          console.log(`package ${packageName} found in db and not readable`);
+          return null;
+        }
+      },
+    },
+    PackageInfo: {
+      __resolveType(obj) {
+        if (isReadablePackage(obj)) {
+          return 'ReadablePackageInfo';
```

Comment:
Are these strings type checked?

Reply:
Yeah, `__resolveType` has to return a `typeof __typename` for the union `PackageInfo`.